File: csharp/SmoothVec2.cs
using System.Numerics;

namespace SmoothKit
{
    public static class SmoothVec2
    {
        public static Vector2 Smooth(Vector2 current, Vector2 target, float speed = 0.1f)
        {
            return new Vector2(
                current.X + (target.X - current.X) * speed,
                current.Y + (target.Y - current.Y) * speed
            );
        }
    }
}

File: ts/smoothVec2.ts
export interface Vec2 {
    x: number;
    y: number;
}

export function smoothVec2(
    current: Vec2,
    target: Vec2,
    speed: number = 0.1
): Vec2 {
    return {
        x: current.x + (target.x - current.x) * speed,
        y: current.y + (target.y - current.y) * speed
    };
}
